### PR TITLE
DPPA-646: Update tiff info method to validate aperture

### PR DESCRIPTION
**Pull request by chris-brent**

### CODE REVIEW

**Comment by jaytarolli** [comment]
For my own sake as I go through and review this, can you explain why we're mapping apertures in the first place? Is the expectation that TIFFs downloaded from MIBItracker will have the letter code stored in the metadata and when loaded with this reader converted to a string representation? Or is it the other way around?

File: mibidata/mibi_image.py [hunk 1]
```diff
@@ -235,28 +235,7 @@ def aperture(self):
...
-                    'required: {}'.format(value,
-                                          _DEPRECATED_APERTURE_MAP[value],
-                                          APERTURE_MAP)
-                )
-            except KeyError:
-                raise ValueError(
-                    'Invalid aperture code \'{}\', must use values'
-                    'from the following map: {}'.format(value, APERTURE_MAP)
-                )
+        self._aperture = MibiImage.validate_aperture(value)
```
Comment (rfinck) [inline mibidata/mibi_image.py]: Why not `self.validate_aperture`

File: mibidata/tiff.py [hunk 2]
```diff
@@ -345,6 +344,9 @@ def _convert_from_previous(description):
         warnings.warn(
             'The "folder" attribute is required if "fov_id" is specified. '
             'Setting "folder" to {}.'.format(description['mibi.folder']))
+    if description.get('mibi.aperture'):
+        description['mibi.aperture'] = mi.MibiImage.validate_aperture(
+            description['mibi.aperture'])
```
Comment (rfinck) [inline mibidata/tiff.py]: This is much nicer than what I had suggested.

**Comment by rfinck** [comment]
@jaytarolli 

> For my own sake as I go through and review this, can you explain why we're mapping apertures in the first place? Is the expectation that TIFFs downloaded from MIBItracker will have the letter code stored in the metadata and when loaded with this reader converted to a string representation? Or is it the other way around?

TIFFs created by the MIBIscope software now use the aperture code, which is the same code that MIBItracker users. This mapping is for backwards compatibility, so that any old ones that contain the string get mapped _to_ the right code. The early access software left the aperture blank in the TIFFs, but some custom ones of ours wrote these strings in and we want them to still work.

**Comment by jaytarolli** [comment]
@rfinck Got it, thanks. I'll update that in the specification doc as well.

File: mibidata/mibi_image.py (hunk 1 above)
Comment (chris-brent) [reply mibidata/mibi_image.py]: Good point, I also didn't like the name so I changed it.

File: mibidata/tiff.py (hunk 2 above)
Comment (chris-brent) [reply mibidata/tiff.py]: 👍

File: mibidata/mibi_image.py [hunk 3]
```diff
@@ -307,7 +307,14 @@ def __str__(self):
         return s
 
     @staticmethod
-    def validate_aperture(value):
+    def parse_aperture(value):
+        """
+        Args:
+            value: A string representing an aperture code.
+        Returns:
+            an aperture code (e.g. 'A' or 'B') matching aperture width
```
Comment (rfinck) [inline mibidata/mibi_image.py]: **A**n aperture code

File: mibidata/mibi_image.py [hunk 4]
```diff
@@ -307,7 +307,14 @@ def __str__(self):
...
+            used during image acquisition.
```
Comment (rfinck) [inline mibidata/mibi_image.py]: Please include a `Raises` section in the docstring.

**Comment by chris-brent** [comment]
@rileysparsons Here's my PR

File: mibidata/mibi_image.py [hunk 5]
```diff
@@ -235,28 +235,7 @@ def aperture(self):
...
-                    'required: {}'.format(value,
-                                          _DEPRECATED_APERTURE_MAP[value],
-                                          APERTURE_MAP)
-                )
-            except KeyError:
-                raise ValueError(
-                    'Invalid aperture code \'{}\', must use values'
-                    'from the following map: {}'.format(value, APERTURE_MAP)
-                )
+        self._aperture = self.parse_aperture(value)
```
Comment (rileysparsons) [inline mibidata/mibi_image.py]: Good stuff!

File: mibidata/mibi_image.py (hunk 4 above)
Comment (chris-brent) [reply mibidata/mibi_image.py]: Done.

**Pull request merged**